agent-csa-bundle/build.gradle.kts

```diff
@@ -1,32 +1,132 @@
+import com.github.jengelman.gradle.plugins.shadow.tasks.ShadowJar
 
 plugins {
   id("maven-publish")
   id("signing")
+  id("com.gradleup.shadow")
 }
 
-dependencies {
-  implementation(project(":agent"))
-}
+// This should be updated for every CSA release, eventually in dependencyManagement?
+
+val csaVersion = "25.4.0-1327"
+val otelInstrumentationVersion: String by rootProject.extra
 
-// This should be updated for every CSA release
-val csaVersion = "25.3.0-1321"
 base.archivesName.set("splunk-otel-javaagent-csa")
 
+val csaReleases: Configuration by configurations.creating {
+  isCanBeResolved = true
+  isCanBeConsumed = false
+}
+
+repositories {
+  ivy {
+    // Required to source artifact directly from github release page
+    // https://github.com/signalfx/csa-releases/releases/download/<version>/oss-agent-mtagent-extension-deployment.jar
+    url = uri("https://github.com/")
+    metadataSources {
+      artifact()
+    }
+    patternLayout {
+      ivy("[organisation]/[module]/releases/download/[revision]/[artifact].[ext]")
+      artifact("[organisation]/[module]/releases/download/[revision]/[artifact].[ext]")
+    }
+  }
+}
+
+dependencies {
+  runtimeOnly(project(":agent", configuration = "shadow"))
+  csaReleases("signalfx:csa-releases:$csaVersion") {
+    artifact {
+      name = "oss-agent-mtagent-extension-deployment"
+      extension = "jar"
+    }
+  }
+}
+
 tasks {
-  val bundleSecureApp by registering(Exec::class) {
-    description = "Bundle the Cisco SecureApp with splunk-otel-java"
-    dependsOn(":agent:assemble")
-    isIgnoreExitValue = false
-    commandLine("./build-agent-csa-fat-jar.sh", version, csaVersion)
-    outputs.file(File("build/splunk-otel-javaagent-csa-$version.jar"))
+
+  // This exists purely as a hack to get the extension jar into our build dir
```

Comment:
I have also been struggling with this. I think the problem is in our mental model where we think that how hard could a simple copy be. From gradle perspective I think archives and files with directories are the same thing so instead of copy we have to tell it to take the files from this archive and place them into another archive. We can hope that internally it is smart enough to somehow translate it to just copying the archive.